File: frontend/src/types/plan.ts
export interface WorkoutExercise {
  name: string;
  sets: number;
  reps: string;
  rest_seconds: number;
  weight_suggestion?: string;
  calories_burned: number;
  notes?: string;
  exercise_completed?: boolean;
}

export interface WorkoutGroup {
  muscle_group: string;
  exercises: WorkoutExercise[];
}

export interface MealOption {
  name: string;
  calories: number;
  protein_g: number;
  carbs_g: number;
  fat_g: number;
  ingredients?: string[];
  recipe_brief?: string;
  platform?: string;
  store_suggestion?: string;
  restaurant_type?: string;
}

export interface MealPlan {
  meal_type: string;
  time: string;
  self_cook: MealOption;
  takeout: MealOption;
  eat_out: MealOption;
}

export interface CalorieSummary {
  bmr: number;
  exercise_burned: number;
  total_intake: number;
  net_calories: number;
}

export interface DailyPlanData {
  motivation_quote: string;
  weather_impact: string;
  training_split: string;
  split_day: string;
  is_rest_day?: boolean;
  warmup: string[];
  workout_groups: WorkoutGroup[];
  cooldown: string[];
  meal_plan: MealPlan[];
  calorie_summary: CalorieSummary;
  recommendations: string[];
}

export interface FitnessPlan {
  id: number;
  user_id: number;
  plan_date: string;
  plan_data: DailyPlanData;
  completed: boolean;
  created_at: string;
  updated_at: string;
}

export interface WeatherInfo {
  temperature: number;
  condition: string;
  humidity: number;
  wind_speed: number;
  location: string;
  icon: string;
}

export interface MotivationInfo {
  quote: string;
  date: string;
}

File: api/app/schemas/plan.py
from pydantic import BaseModel, Field
from typing import Dict, Any, Optional, List
from datetime import date, datetime

class FitnessPlanBase(BaseModel):
    plan_date: date = Field(..., description="计划日期")
    plan_data: Dict[str, Any] = Field(..., description="计划内容 (JSON)")

class FitnessPlanCreate(FitnessPlanBase):
    pass

class FitnessPlanGenerate(BaseModel):
    plan_date: date = Field(..., description="计划日期")

class FitnessPlanUpdate(BaseModel):
    completed: Optional[bool] = None
    plan_data: Optional[Dict[str, Any]] = None

class FitnessPlanResponse(FitnessPlanBase):
    id: int
    user_id: int
    completed: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

# AI Plan Structure
class WorkoutExercise(BaseModel):
    name: str
    sets: int
    reps: str
    rest_seconds: int
    weight_suggestion: Optional[str] = None
    calories_burned: int
    notes: Optional[str] = None

class WorkoutGroup(BaseModel):
    muscle_group: str
    exercises: list[WorkoutExercise]

class MealOption(BaseModel):
    name: str
    calories: int
    protein_g: float
    carbs_g: float
    fat_g: float
    ingredients: Optional[list[str]] = None
    recipe_brief: Optional[str] = None
    platform: Optional[str] = None
    store_suggestion: Optional[str] = None
    restaurant_type: Optional[str] = None

class MealPlan(BaseModel):
    meal_type: str
    time: str
    self_cook: MealOption
    takeout: MealOption
    eat_out: MealOption

class DailyPlan(BaseModel):
    motivation_quote: str
    weather_impact: str
    training_split: str
    split_day: str
    warmup: list[str]
    workout_groups: list[WorkoutGroup]
    cooldown: list[str]
    meal_plan: list[MealPlan]
    total_calories_burned: int
    total_calories_intake: int
    recommendations: list[str]
